libs/features/mas/commerce/src/literals.js

```diff
@@ -1,29 +1,17 @@
-import { ERROR_MESSAGE_MISSING_LITERALS_URL } from './constants.js';
 import { Defaults } from './defaults.js';
 import { equalsCaseInsensitive } from './external.js';
 
-function generateLiteralsPromise(settings) {
-    if (!settings.priceLiteralsURL) {
-        throw new Error(ERROR_MESSAGE_MISSING_LITERALS_URL);
-    }
-    return new Promise((resolve) => {
-        window.fetch(settings.priceLiteralsURL).then((response) => {
-            response.json().then(({ data }) => {
-                resolve(data);
-            });
-        });
-    });
-}
-
 /**
+ * Method resolves price literals for the given language from the group of price literals.
+ * That group is either imported from json file or it is received as a parameter (in case of unit tests).
+ *
  * @param {Commerce.Price.Settings} settings
+ * @param priceLiterals
  * @returns {Promise<Record<string, string>>}
  */
-export async function fetchPriceLiterals(settings) {
-    const priceLiteralsPromise =
-        settings.priceLiteralsPromise || generateLiteralsPromise(settings);
+export async function getPriceLiterals(settings, priceLiterals) {
     //we are expecting an array of objects with lang and literals
-    const data = await priceLiteralsPromise;
+    const { data } = priceLiterals ? priceLiterals : await import('../price-literals.json');
```

Comment:
I am a bit confused here; the description says:
>This change increases commerce.js for 5KB **but eliminates the call for price-literals.json** and hence eliminates the possibility of its failure.

If we're still calling price-literals, why aren't we taking it from content?

Reply:
@narcis-radu  We don't fetch `price-literals.json` any more, this is import, not fetch. There is no call `price-literals.json` ay more.
We are injecting it from source code into `commerce.js` with `import('../price-literals.json')` if it's not already provided with `priceLiterals` argument, which will be the case only for unit tests. In unit tests we cannot use `import` so I had to read it from source with `readFile` and then send it as an argument to `getPriceLiterals()`.